save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/config/KafkaConfiguration.kt

```diff
@@ -0,0 +1,164 @@
+package com.saveourtool.save.orchestrator.config
+
+import com.saveourtool.save.kafka.TestExecutionTaskDto
+import com.saveourtool.save.orchestrator.kafka.AgentKafkaListener
+import com.saveourtool.save.orchestrator.kafka.KafkaSender
+import org.apache.kafka.clients.admin.NewTopic
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.consumer.CooperativeStickyAssignor
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
+import org.springframework.kafka.core.*
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer
+import org.springframework.kafka.listener.SeekToCurrentErrorHandler
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer
+import org.springframework.kafka.support.serializer.JsonDeserializer
+import org.springframework.kafka.support.serializer.JsonSerializer
+import java.util.*
+
+/**
+ * Kafka producer and consumer configuration
+ *
+ * @property kafkaProperties
+ */
+@Configuration
+@Profile("dev & kafka")
+internal class KafkaConfiguration(
+    private val kafkaProperties: KafkaProperties) {
+    /**
+     * @property requestTopicName
+     */
+    @Value("\${kafka.test.execution.request.topic.name}")
+    lateinit var requestTopicName: String
+
+    /**
+     * @property requestTopicNameDlt
+     */
+    @Value("\${kafka.test.execution.request.topic.name}.DLT")
+    lateinit var requestTopicNameDlt: String
+
+    /**
+     * @property responseTopicName
+     */
+    @Value("\${kafka.test.execution.response.topic.name}")
+    lateinit var responseTopicName: String
+
+    /**
+     * @property consumerGroup
+     */
+    @Value("\${spring.kafka.consumer.group-id}")
+    lateinit var consumerGroup: String
+
+    /**
+     * @return kafka producer properties
+     */
+    @Bean
+    fun producerConfig(): Map<String, Any?> {
+        val producerProps = HashMap(kafkaProperties.buildProducerProperties())
+        producerProps[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = StringSerializer::class.java as Object
+        producerProps[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = JsonSerializer::class.java as Object
```

Comment:
Currently we are using `kotlinx.serialization` on both agent and orchestrator to ensure that they exchange data in the same format (e.g. Jackson and kx.serialization have different field for type information in case of polymorphic serialization, may serialize DateTime differently etc.). As I understarnd, `org.springframework.kafka.support.serializer.JsonSerializer` uses Jackson. Can we - and will it be worth it - use kx.serialization here?

Reply:
yes but it will be separate task, on save-agent we should use only kx.serialization in fact. but it would be cool it it will be compartible with spring json serializer because it is widely used in the community, it is very flexible serializer.